src/services/GroupService.ts

```diff
@@ -0,0 +1,48 @@
+import { ModelCtor } from 'sequelize';
+import { v4 as uuidv4 } from 'uuid';
+import { GroupModel } from '../models/GroupModel';
+import { UserModel } from '../models/UserModel';
+import { GroupAttributes, GroupInstance, Permission } from '../types';
+
+class GroupService {
+  private groupModel: ModelCtor<GroupInstance>;
+
+  constructor (groupModel: ModelCtor<GroupInstance>) {
+    this.groupModel = groupModel;
+  }
+
+  getGroups = async (): Promise<GroupAttributes[]> => (
+    await this.groupModel.findAll({
+      attributes: ['name', 'permission', 'group_id'],
+      include: UserModel
+    })
+  )
+
+  findGroup = async (id: string): Promise<GroupAttributes | undefined> => (
+    await (await this.groupModel.findByPk(id, {
```

Comment:
i suggest splitting it for more readability

Reply:
get rid of extra await